src/Controller/DefaultController.php

```diff
@@ -18,12 +18,20 @@ class DefaultController extends AbstractController
 {
     /**
      * @Route("/", name="app_home", methods={"GET"})
-     * @Route("/{path}", name="app_default", methods={"GET"}, requirements={"path"="^(?!build).+"})
+     * @Route("/explore", name="app_explore", methods={"GET"})
      */
     public function index(DirectoryListing $listing, Request $request, NextChapterResolver $resolver, string $mangaRoot, PaginatorInterface $paginator): Response
     {
-        $path = $request->attributes->get('path', '');
-        $target = sprintf('%s/%s', $mangaRoot, $path);
+        if ('app_home' === $request->attributes->get('_route')) {
+            return $this->redirectToRoute('app_explore');
+        }
+
+        $path = (string) $request->query->get('path', '/');
+        $decodedPath = rawurldecode($path);
+        if (str_ends_with($decodedPath, '.zip')) {
```

Comment:
Need to handle `cbz,epub` extension with posibility of case sensitive. `str_ends_with` is available on php 8.0, it fails psalm analytics on php 7.2.

Reply:
Missing redirect test case.